### PR TITLE
CONSECTIVE TRANSFORM

### PR DESCRIPTION
**Pull request by aertoria**
CONSECTIVE TRANSFORM

### CODE REVIEW

**Comment by aertoria** [comment]
CONSECTIVE: Returns all datapoint's for which the interval between adjacent datapoints is less than or equal to the specified time interval, for consective specified time threshold

USAGE:
CONSECTIVE(-1h:argus.jvm:mem.heap.used:avg,$20m,$1m)

@saaja-sfdc

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/ConsecutiveValueMapping.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import com.salesforce.dva.argus.service.metric.MetricReader;
+import com.salesforce.dva.argus.system.SystemAssert;
+
+/**
+ * CONSECUTIVE: Returns all datapoint's for which the interval between adjacent datapoints is less than or equal to the specified time interval, for consective specified time threshold .
```
Comment (saaja-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/ConsecutiveValueMapping.java]: Remove 'CONSECUTIVE:'

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/ConsecutiveValueMapping.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ * @author  Ethan Wang (ethan.wang@salesforce.com)
+ */
+public class ConsecutiveValueMapping implements ValueMapping {
+	private long threshold;
+	private long connectDistance;
+	private ArrayList<Long> keyList;
+	private ArrayList<Long> resultKeyList;
+	
+	@Override
+	public Map<Long, String> mapping(Map<Long, String> originalDatapoints, List<String> constants) {
```
Comment (saaja-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/ConsecutiveValueMapping.java]: Needs javadoc, specifically for the parameters, how they affect the operation, etc...

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/TransformFactory.java [hunk 3]
```diff
@@ -236,6 +238,7 @@ public Transform getTransform(String functionName) {
         FILL_CALCULATE("FILL_CALCULATE", "Creates a constant line based on the calculated value."),
         INCLUDE("INCLUDE", "Retains metrics based on the matching of a regular expression against the metric name."),
         EXCLUDE("EXCLUDE", "Culls metrics based on the matching of a regular expression against the metric name."),
+        CONSECUTIVE("CONSECUTIVE","Filter out all values that is not consecutive"),
```
Comment (saaja-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/TransformFactory.java]: Filters out all values that are non-consecutive.

**Comment by saaja-sfdc** [comment]
@aertoria , just some minor documentation changes, etc...   I think once these are fixed, we can merge this.  Thanks!

**Comment by aertoria** [comment]
@saaja-sfdc Thanks for reading the code. I will handle those suggestions right away.

**Comment by aertoria** [comment]
@saaja-sfdc Here you go. Thanks.

**Pull request merged**